### PR TITLE
Tyle8552/raster

### PR DESCRIPTION
**Pull request by ghost**
Status: Ready

Initial implementations for 4 raster samples.

Make sure to download new raster folder in samples-data.

### CODE REVIEW

**Comment by ghost** [comment]
@john0005 If you get a chance, could you review? Still need to add readmes and images but the rest is ready.

**Comment by ghost** [comment]
@vGandhi I think JC is on vacation. If you get a change, could you do a review of this?

File: src/main/java/com/esri/samples/raster/blend_renderer/BlendRendererController.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  /**
+   * Updates the raster layer renderer according to the chosen property values.
+   */
+  public void updateRenderer() {
+
+    ColorRamp colorRamp = colorRampComboBox.getSelectionModel().getSelectedItem() != ColorRamp.PresetType.NONE?
+        new ColorRamp(colorRampComboBox.getSelectionModel().getSelectedItem(), 800) : null;
+
+    // if color map is not NONE, color the hillshade elevation raster instead of using satellite imagery raster color
```
Comment (vGandhi) [inline src/main/java/com/esri/samples/raster/blend_renderer/BlendRendererController.java]: if color ```ramp``` is not NONE ?

File: src/main/java/com/esri/samples/raster/blend_renderer/BlendRendererSample.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+public class BlendRendererSample extends Application {
+
+  private static BlendRendererController controller;
+
+  @Override
+  public void start(Stage stage) throws IOException {
+    // set up the scene
+    FXMLLoader loader = new FXMLLoader(getClass().getResource
+        ("/fxml/blend_renderer.fxml"));
```
Comment (vGandhi) [inline src/main/java/com/esri/samples/raster/blend_renderer/BlendRendererSample.java]: Could this fit in the line above?

**Comment by ghost** [comment]
@vGandhi Good catches, thanks. I've addressed them.

File: src/main/java/com/esri/samples/raster/hillshade_renderer/HillshadeRendererSample.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+public class HillshadeRendererSample extends Application {
+
+  private static HillshadeRendererController controller;
+
+  @Override
+  public void start(Stage stage) throws IOException {
+    // set up the scene
+    FXMLLoader loader = new FXMLLoader(getClass().getResource
+        ("/fxml/hillshade_renderer.fxml"));
```
Comment (vGandhi) [inline src/main/java/com/esri/samples/raster/hillshade_renderer/HillshadeRendererSample.java]: Could this fit in the line above?

File: src/main/java/com/esri/samples/raster/rgb_renderer/RgbRendererController.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  }
+
+  void terminate() {
+
+    if (mapView != null) {
+      mapView.dispose();
+    }
+  }
+
+}
```
Comment (vGandhi) [inline src/main/java/com/esri/samples/raster/rgb_renderer/RgbRendererController.java]: Missing newline.

File: src/main/resources/fxml/blend_renderer.fxml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                      GridPane.columnIndex="1"/>
+            <Label text="Azimuth:" GridPane.rowIndex="2" GridPane.columnIndex="0"/>
+            <Slider fx:id="azimuthSlider" max="360.0" showTickLabels="true"
+                    majorTickUnit="60.0" value="90.0" GridPane.rowIndex="2" GridPane.columnIndex="1"/>
+            <Label text="Altitude:" GridPane.rowIndex="3" GridPane.columnIndex="0"/>
+            <Slider fx:id="altitudeSlider" max="90.0" value="30.0" showTickLabels="true"
+                    majorTickUnit="15.0" GridPane.rowIndex="3" GridPane.columnIndex="1"/>
+        </GridPane>
+    </VBox>
+</StackPane>
```
Comment (vGandhi) [inline src/main/resources/fxml/blend_renderer.fxml]: Missing newline in FXML files.

**Comment by vGandhi** [comment]
@tyle8552 Looks good.

I noticed that the controllers use public fields omitting the @FXML. Existing controllers use @FXML with private fields. I think it is best to follow one of the patterns.

**Comment by ghost** [comment]
@vGandhi made the changes you suggested

**Pull request merged**